matchtree.go

```diff
@@ -321,10 +282,57 @@ func (t *fileNameMatchTree) prepare(doc uint32) {
 	t.child.prepare(doc)
 }
 
-func (t *substrMatchTree) prepare(nextDoc uint32) {
-	t.matchIterator.prepare(nextDoc)
+func (t *substrMatchTree) prepare(doc uint32) {
+	t.matchIterator.prepare(doc)
 	t.current = t.matchIterator.candidates()
 	t.contEvaluated = false
+
+	var fileStart uint32
+	if doc > 0 {
+		fileStart = t.fileEndRunes[doc-1]
+	}
+
+	// enrich candidates with ctags information
```

Comment:
I worry that doing this in substrMatchTree will slow down zoekt. This is a lot of extra work. Additionally a lot of our substrMatchTree occur in a `no collect` which means we will never call fillmatches on it. I wonder if this should be work we do during match collection rather than in prepare.

Reply:
I moved the logic enrigh matches with symbol data from `prepare` to `fillContentMatches`. The overhead should be much smaller now. I also moved the call to `findSection` from `matchScore` up to `fillContentMachtes` so that we only have to call it once.